### PR TITLE
Collapsible meeting sidebar [ready]

### PR DESCRIPTION
**Pull request by ackernaut**
### Changes
- Adds a toggle that opens/closes sidebar for new meeting
- Some meeting views are more responsive/adaptive
- Start moving some constants out of `ui.js`
- A few things updated to use `react-emotion`
- Only updates old meeting when there are shared components with new meeting that got some updates
- Still favoring small laptops and up (sidebar is default to open)

### To test
- [x] Tap team name in new meeting sidebar to copy link to meeting
- [x] Tapping logo in sidebar now goes to My Dashboard
- [x] Can toggle to open and close new meeting sidebar
- [x] There is a tappable backdrop for narrow viewports when the sidebar menu is open
- [x] Tapping a sidebar nav item closes the menu automatically for narrow viewports
- [ ] Probably a good idea to run a retro and action meeting with 2 local players

### To review
- [ ] `CopyLink` abstracted and used with updated `CopyShortLink`
- [ ] Style constants for meeting context removed from `ui.js`
- [ ] Deprecates breakpoints from `ui.js` in favor of `breakpoints.js` (not perfect but a step forward)
- [ ] A simple way to get window size `getWindowSize.js` or import Yet Another Package™?
- [ ] Handlers for sidebar toggles (toggle in sidebar, toggle in heading, sidebar clicks close for small viewports)

### To do
- [x] Created an issue: sidebar should default to closed at the small sidebar breakpoint (#2257)
- [x] Created an issue: retro card grid layout should update when the sidebar is toggled (#2256)

### CODE REVIEW

**Comment by ackernaut** [comment]
@mattkrick sir this is ready for your review! I left a list of things above for consideration.

**Comment by mattkrick** [comment]
this is amazing! really, really solid work. just a couple little code cleanups.
Only 1 tiny thing I'd like to see:

- [ ] on a small viewport, clicking blank space on the sidebar does not close the sidebar

nice work!

File: src/universal/components/RetroReflectPhase/RetroReflectPhase.js [hunk 1]
```diff
@@ -15,6 +15,13 @@ import ScrollableBlock from 'universal/components/ScrollableBlock'
 import MeetingPhaseWrapper from 'universal/components/MeetingPhaseWrapper'
 import FlatButton from 'universal/components/FlatButton'
 import IconLabel from 'universal/components/IconLabel'
+import styled from 'react-emotion'
+
+const StyledWrapper = styled(MeetingPhaseWrapper)({
+  // TODO: base wrapper on min-width of cards, not this styled div (TA)
+  minWidth: '48rem',
```
[the remark this answers is not in the document]
Comment (ackernaut) [reply src/universal/components/RetroReflectPhase/RetroReflectPhase.js]: I think I’m still accounting for some gutters here, but I’ll see if there is something better I can do with this.

File: src/universal/components/CopyLink.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    super(props)
+    this.confirmationTimeout = null
+  }
+
+  state = {
+    confirmingCopied: false
+  }
+
+  componentWillUnmount () {
+    this.clearConfirmationTimeout()
```
[the remark this answers is not in the document]
Comment (ackernaut) [reply src/universal/components/CopyLink.js]: Changing this doesn’t seem to work. Moving `clearTimeout` with or without the wrapping `if` statement to `componentWillUnmount` and we don’t get a confirmation. Going to leave for now.

**Comment by mattkrick** [comment]
>  it should close on any click in the sidebar area

i don't think this is standard behavior. every other app out there (google docs, google maps, reddit, slack, etc.) lets you tap the negative space on the sidebar & nothing happens. to close, you either have to hit a link/button, hit the collapse button, or hit the overlay.

File: src/client/clientSchema.graphql [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
 extend type RetrospectiveMeeting {
   localPhase: NewMeetingPhase!
   localStage: NewMeetingStage!
-  sidebarCollapsed: Boolean
   reflectionsInFlight: [RetroReflection]
 }
 
 extend type Team {
   contentFilter: String
+  sidebarCollapsed: Boolean
```
Comment (ackernaut) [inline src/client/clientSchema.graphql]: `sidebarCollapsed` is only used in the new meeting context at the moment. Should I rename this to `meetingSidebarCollapsed`?

Comment (mattkrick) [reply src/client/clientSchema.graphql]: yeah, `isMeetingSidebarCollapsed` gets my vote. i'm always a fan of long but clear var names, and booleans that start with a question

**Comment by ackernaut** [comment]
@jordanh hrm 🤔 I cannot reproduce that locally…any useful messages in the console? did you pull down my last commit(s)?

**Comment by jordanh** [comment]
# Test Cycle 2
## Status: completed

## Tests
- [x] Tap team name in new meeting sidebar to copy link to meeting
- [x] Tapping logo in sidebar now goes to My Dashboard
- [x] Can toggle to open and close new meeting sidebar
- [x] There is a tappable backdrop for narrow viewports when the sidebar menu is open
- [x] Tapping a sidebar nav item closes the menu automatically for narrow viewports
- [x] Probably a good idea to run a retro and action meeting with 2 local players

**Comment by jordanh** [comment]
Worked on dev! Hmm...makes me wonder about what might be going on locally...

**Pull request merged**